app/assets/javascripts/Components/submission_file_manager.jsx

```diff
@@ -254,9 +256,52 @@ class SubmissionFileManager extends React.Component {
     );
   };
 
+  renderRequiredFiles = () => {
+    let requiredFilesBox;
+
+    if (this.state.requiredFiles.length > 0) {
+      requiredFilesBox = (
+        <div>
+          <p>{I18n.t("assignments.assignment_files")}</p>
+          {this.state.requiredFiles.map(filename => {
+            return (
+              <div key={filename}>
+                <input
+                  value={filename}
+                  type={"checkbox"}
+                  disabled={true}
+                  checked={this.state.files.some(element => element.key === filename)}
+                />
+                <span> {filename}</span>
+              </div>
+            );
+          })}
+        </div>
+      );
+    } else {
+      requiredFilesBox = <div></div>;
+    }
+    return (
+      <div className={"pane-wrapper small-bottom-margin"}>
+        <div className={"pane"}>
+          {requiredFilesBox}
```

Comment:
The other comment I have about presentation is that there's another flash message that appears when students upload a file that says "You still need to submit X required file(s) for this assignment" with a list. I don't think we need that flash message any more, since it'll be more obvious now which files still need to be submitted. *However* it would be good to use the message text in this component; in particular I like that it includes a count of the number of files that still need to be submitted. I'd also like to see a new message that appears if all required files have been submitted (e.g. "Good job! ..." etc.).